server.js

```diff
@@ -14,6 +14,38 @@ const sequelize = new Sequelize('hamishgpt', 'user', 'root', {
   dialect: 'mysql'
 });
 
+app.get('/ping', function(request, response) {
+  response.send("pong");
+});
+
+app.post('/registration', async (request, response) => {
+  const { username, email, password } = request.body;
+  if (username === null || username.length < 1) {
+    return response.status(400).send('Username is invalid');
+  }
+  if (email === null || email.length < 1) {

```

Comment:
You are not checking if it's a valid email address. You are just checking if the string for the email is at least 1 character.

Reply:
fixed